### ISSUE TITLE
Account compilation support

### ISSUE THREAD

**Issue by greenhat** [0xPolygonMiden/compiler#139]
This is a tracking issue for everything needed for a user to compile an account code.

### Goal

Compile an account code in Rust (like basic wallet example). The account code should be able to use the kernel tx API and stdlib.

```[tasklist]
### Tasks

```

**Comment by bitwalker** [comment]
@greenhat I think this issue overlaps a bit too much with #141, as they have virtually identical requirements. Some dependencies of being able to implement proper account compilation require packaging, so there isn't really a distinction between supporting account compilation and supporting account compilation + packaging.

We _could_ modify this issue to be about publishing/distributing account/note script artifacts, and the integration with the Miden client, but it might be better to wait until we get closer to implementation to start tracking the requirements.

**Comment by greenhat** [comment]
> @greenhat I think this issue overlaps a bit too much with #141, as they have virtually identical requirements. Some dependencies of being able to implement proper account compilation require packaging, so there isn't really a distinction between supporting account compilation and supporting account compilation + packaging.

I agree. A packaging is required at account stage, since we cannot just bolt it on later when we start working on note scripts.

> We _could_ modify this issue to be about publishing/distributing account/note script artifacts, and the integration with the Miden client, but it might be better to wait until we get closer to implementation to start tracking the requirements.

I agree, local deps is the way to go for the MVP.

I added the tasks to this issue that I think we're missing for MVP and are "note-only". I'm fine with merging it with #141.
